sway-lib-std/src/identity.sw

```diff
@@ -21,3 +24,53 @@ impl core::ops::Eq for Identity {
         }
     }
 }
+
+impl Identity {
```

Comment:
Both methods should probably return `Option` instead of `Result`. There is no reason to assume in the type interface that there are only two possibilities. And proper types shouldn't be errors.